gnocchi/indexer/sqlalchemy.py

```diff
@@ -619,7 +619,9 @@ def create_archive_policy(self, archive_policy):
     def list_archive_policy_rules(self):
         with self.facade.independent_reader() as session:
             return session.query(ArchivePolicyRule).order_by(
-                ArchivePolicyRule.metric_pattern.desc()).all()
+                ArchivePolicyRule.metric_pattern.asc(),
```

Comment:
should we keep this descending? i think it's best if it matches the most specific pattern before the more generic pattern.

ie. match disk.io.* before disk.* or *

Reply:
Of course, me stupid.